File: be_hotel/src/main/java/no/services/booking/BookingService.java
package no.services.booking;

import no.modal.entity.Booking;
import no.modal.request.BookingRequest;
import org.springframework.data.domain.Page;

public interface BookingService {
    // post
    Booking createBooking(BookingRequest bookingRequest);
    // get
    Page<Booking> getBookings(int page, int size);
    Booking findById(long id);
    // put
    Booking updateBooking(BookingRequest bookingRequest, long id);
    //delete
    boolean deleteBooking(long id);

}

File: fe_hotel/web_app/src/services/BookingService.js
import axiosInstance from "./axiosConfigs";

const BookingService = {
    create(body){
        return axiosInstance.post('Booking/createBooking', body)
    },
    getAll(page, size){
        return axiosInstance.get(`Booking/getAllBookings/${page}/${size}`)
    },
    getId(id){
        return axiosInstance.get(`Booking/findById/${id}`)
    },
    update(id, body){
        return axiosInstance.put(`Booking/updateBooking/${id}`, body)
    },
    delete(id){
        return axiosInstance.get(`Booking/deleteBooking/${id}`)
    }
}
export default BookingService;